services/src/main/java/io/grpc/protobuf/services/ProtoReflectionService.java

```diff
@@ -16,524 +16,100 @@
-        }
-      }
-
-      // Replace the index if the underlying mutable services have changed. Check both the file
-      // descriptors and the service names, because one file descriptor can define multiple
-      // services.
-      FileDescriptorIndex mutableServicesIndex = index.getMutableServicesIndex();
-      if (!mutableServicesIndex.getServiceFileDescriptors().equals(serverFileDescriptors)
-          || !mutableServicesIndex.getServiceNames().equals(serverServiceNames)) {
-        index =
-            new ServerReflectionIndex(server.getImmutableServices(), serverMutableServices);
-        serverReflectionIndexes.put(server, index);
-      }
-
-      return index;
-    }
-  }
-
   @Override
-  public StreamObserver<ServerReflectionRequest> serverReflectionInfo(
-      final StreamObserver<ServerReflectionResponse> responseObserver) {
-    final ServerCallStreamObserver<ServerReflectionResponse> serverCallStreamObserver =
-        (ServerCallStreamObserver<ServerReflectionResponse>) responseObserver;
-    ProtoReflectionStreamObserver requestObserver =
-        new ProtoReflectionStreamObserver(getRefreshedIndex(), serverCallStreamObserver);
-    serverCallStreamObserver.setOnReadyHandler(requestObserver);
-    serverCallStreamObserver.disableAutoRequest();
-    serverCallStreamObserver.request(1);
-    return requestObserver;
+  public ServerServiceDefinition bindService() {
+    return createServerServiceDefinition();
   }
 
-  private static class ProtoReflectionStreamObserver
-      implements Runnable, StreamObserver<ServerReflectionRequest> {
-    private final ServerReflectionIndex serverReflectionIndex;
-    private final ServerCallStreamObserver<ServerReflectionResponse> serverCallStreamObserver;
-
-    private boolean closeAfterSend = false;
-    private ServerReflectionRequest request;
-
-    ProtoReflectionStreamObserver(
-        ServerReflectionIndex serverReflectionIndex,
-        ServerCallStreamObserver<ServerReflectionResponse> serverCallStreamObserver) {
-      this.serverReflectionIndex = serverReflectionIndex;
-      this.serverCallStreamObserver = checkNotNull(serverCallStreamObserver, "observer");
-    }
-
-    @Override
-    public void run() {
-      if (request != null) {
-        handleReflectionRequest();
-      }
-    }
+  private <ReqT, RespT> ServerServiceDefinition createServerServiceDefinition() {
+    ServerServiceDefinition serverServiceDefinitionV1 = ProtoReflectionServiceV1.newInstance()
+        .bindService();
+    MethodDescriptor<ReqT, RespT> methodDescriptorV1 = (MethodDescriptor<ReqT, RespT>) serverServiceDefinitionV1.getServiceDescriptor()
+        .getMethods().iterator().next();
+    MethodDescriptor<ReqT, RespT> methodDescriptorV1Alpha = methodDescriptorV1.toBuilder()
+        .setFullMethodName(generateFullMethodName(SERVICE_NAME, METHOD_NAME))
+        .setSchemaDescriptor(
+            new ServerReflectionMethodDescriptorSupplier(METHOD_NAME))
```

Comment:
It is probably worth it to copy these from the v1alpha generated code, by get()ing it from the method descriptor.

Reply:
If you mean to first create a v1alpha ServerServiceDefinition and get the method descriptor from it, since the v1alpha generated stub has the following 
`public static final io.grpc.ServerServiceDefinition bindService(AsyncService service) {`

And we don't have any implementation of this v1alpha AsyncService, so we can't create such a ServerServiceDefinition.

Reply:
Done.

Reply:
I see you copied the method name, but not the schema descriptor, here and for the ServiceDescriptor.

Reply:
Done.